worker/uniter/runner/context/context.go

```diff
@@ -1023,14 +1035,14 @@ func (ctx *HookContext) handleReboot(ctxErr error) error {
 		if ctxErr != nil {
 			return ctxErr
 		}
-		return ErrReboot
+		ctxErr = ErrReboot
 	case jujuc.RebootNow:
-		return ErrRequeueAndReboot
+		ctxErr = ErrRequeueAndReboot
 	}
 
-	// Do a best-effort attempt to set the unit status; we don't care if it
-	// fails as we will request a reboot anyway.
-	if err := ctx.unit.SetUnitStatus(status.Rebooting, "", nil); err != nil {
+	// Do a best-effort attempt to set the unit agent status; we don't care
+	// if it fails as we will request a reboot anyway.
+	if err := ctx.unit.SetAgentStatus(status.Rebooting, "", nil); err != nil {
```

Comment:
Original code (before refactoring) always failed (logged but otherwise ignored the error) trying to set the status because `status.Rebooting` is not a valid workload status.